crates/task-impls/src/helpers.rs

```diff
@@ -84,9 +100,14 @@ pub(crate) async fn fetch_proposal<TYPES: NodeType, V: Versions>(
 
             // Then, if it's `Some`, make sure that the data is correct
             if let Some(hs_event) = event.as_ref() {
-                if let HotShotEvent::QuorumProposalResponseRecv(_view_number, quorum_proposal) =
+                if let HotShotEvent::QuorumProposalResponseRecv(quorum_proposal) =
                     hs_event.as_ref()
                 {
+                    // Make sure that the quorum_proposal is valid
+                    if quorum_proposal.validate_signature(&mem).is_err() {
+                        return None
+                    }
+
```

Comment:
Maybe we should just keep waiting or else we could just be attacked by giving bad proposal responses from anyone and end up ignoring the real response

Reply:
Done